opentelemetry-otlp/src/exporter/http/mod.rs

```diff
@@ -316,46 +316,17 @@ fn resolve_endpoint(
 
 #[allow(clippy::mutable_key_type)] // http headers are not mutated
 fn add_header_from_string(input: &str, headers: &mut HashMap<HeaderName, HeaderValue>) {
-    for pair in input.split_terminator(',') {
-        if pair.trim().is_empty() {
-            continue;
-        }
-        if let Some((k, v)) = pair.trim().split_once('=') {
-            if !k.trim().is_empty() && !v.trim().is_empty() {
-                if let (Ok(key), Ok(value)) = (
-                    HeaderName::from_str(k.trim()),
-                    HeaderValue::from_str(v.trim()),
-                ) {
-                    headers.insert(key, value);
-                }
-            }
+    for (key, value) in parse_header_string(input) {
+        if let (Ok(key), Ok(value)) = (HeaderName::from_str(key), HeaderValue::from_str(value)) {
+            headers.insert(key, value);
```

Comment:
Both `http` and `tonic` modules are working with `HeaderName`/`HeaderValue` so I just pushed some updates to simplify the code of `parse_header_string` to return `impl Iterator<Item = (HeaderName, HeaderValue)> + '_` (instead of `
impl Iterator<Item = (&str, &str)> + '_
`)

I updated `http` module applying your suggestion (function is very light now but I didn't remove it so as not to affect unit tests)
```rust
fn add_header_from_string(input: &str, headers: &mut HashMap<HeaderName, HeaderValue>) {
    headers.extend(parse_header_string(input));
}
```